src/hct_mis_api/apps/payment/services/payment_plan_services.py

```diff
@@ -372,7 +373,8 @@ def check_payment_plan_and_update_status(self, approval_process: ApprovalProcess
             if approval_type == Approval.FINANCE_RELEASE:
                 self.payment_plan.status_mark_as_reviewed()
                 notification_action = PaymentPlan.Action.REVIEW
-                # remove imported and export files
+                # AB#272790
+                transaction.on_commit(lambda: update_exchange_rate_on_release_payments.delay(str(self.payment_plan.id)))
```

Comment:
So basically, someone approves PP with USD values based on the historical exchange rate, and all of these values are being recalculated immediately after approval? Sounds like FINANCE_RELEASE decision is being made based on stale data, is this flow correct?

Reply:
`Finance person is expected to release based on the amount in the currency of the payment plan. Exchange rate is purely indicative.`